### PR TITLE
Refactor runtime and http service

### PR DESCRIPTION
**Pull request by adzialocha**
This PR ..

* [x] Moves `axum` http server into managed service
* [x] Uses graceful shutdown feature of `axum`
* [x] Replaces `TaskManager` with new `ServiceManager`
* [x] Makes the application shut down when a service returns an error or panics
* [x] Refactors file structure to prepare for more "service-centric" architecture

Closes: #85 and partly #87 

## 📋 Checklist

- [x] Add tests that cover your changes
- [x] Add this PR to the _Unreleased_ section in `CHANGELOG.md`
- [x] Link this PR to any issues it closes
- [x] New files contain a SPDX license header

### CODE REVIEW

File: aquadoggo/src/server.rs [hunk 1]
```diff
@@ -64,33 +35,40 @@ pub fn build_server(state: ApiState) -> Router {
...
+
+    axum::Server::try_bind(&http_address)?
+        .serve(build_server(context).into_make_service())
+        .with_graceful_shutdown(async {
+            signal.await.ok();
+        })
         .await?;
+
     Ok(())
 }
```
Comment (sandreae) [inline aquadoggo/src/server.rs:55]: Just checking, but this is the actual service function we add to the service manager, all services will be passed as functions with the same signature as this. Correct?

File: aquadoggo/src/service_manager.rs [hunk 2]
```diff
@@ -62,29 +51,92 @@ where
...
+}
+
+// Service manager for orchestration of long-running concurrent processes.
+//
+// This manager offers a message bus between services for cross-service communication. It also
+// sends a shutdown signal to allow services to react to it gracefully.
+//
+// Stopped services (because of a panic, error or successful return) will send an exit signal which
+// can be subscribed to via the `on_exit` method. Usually stopped services indicate system failure
+// and it is recommended to stop the application when this events occurs.
```
Comment (sandreae) [inline aquadoggo/src/service_manager.rs:114]: Cool, so this is the "stop everything if one service fails" pattern we just discussed in the previous PR?

File: aquadoggo/src/service_manager.rs [hunk 3]
```diff
@@ -195,34 +282,77 @@ mod tests {
...
         });
 
         manager.shutdown().await;
 
         // Check if we received the message in all services
         assert_eq!(counter.load(Ordering::Relaxed), 5);
     }
+
+    #[tokio::test]
+    async fn on_exit() {
```
Comment (sandreae) [inline aquadoggo/src/service_manager.rs:317]: :+1:

**Review by sandreae — APPROVED** [review]
Looks great, you already added the `exit_signal` :smile:

File: aquadoggo/src/server.rs (hunk 1 above)
Comment (adzialocha) [reply aquadoggo/src/server.rs:55]: Yes exactly!

File: aquadoggo/src/service_manager.rs (hunk 2 above)
Comment (adzialocha) [reply aquadoggo/src/service_manager.rs:114]: Correct, the service manager in itself does not enforce this, but it gives you the option.

The aquadoggo cli `main.rs` uses that pattern and moves over to shutdown as soon as CTRL+C has been pressed or a service stopped aka the exit signal fired.

**Pull request merged**